src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -155,6 +160,35 @@ public void close() {
     }
   }
 
+  /**
+   * Determine the acceptable means of authentication for a given DRS ID, including the passport
+   * issuers when supported.
+   *
+   * @param drsObjectId the object ID for which to look up authorizations
+   * @return the `DrsAuthorizations` for this ID
+   * @throws IllegalArgumentException if there is an issue with the object id
+   * @throws SnapshotNotFoundException if the snapshot for the DRS object cannot be found
+   */
+  public DRSAuthorizations lookupAuthorizationsByDrsId(String drsObjectId) {
+    DRSAuthorizations auths = new DRSAuthorizations();
+
+    auths.addSupportedTypesItem(DRSAuthorizations.SupportedTypesEnum.BEARERAUTH);
+    // TODO: add to bearer_auth_issuers, ask Muscles.
+
+    SnapshotCacheResult snapshot = lookupSnapshotForDRSObject(drsObjectId);
```

Comment:
I don't *think* that we needed to wrap with a `try (DrsRequestResource r = new DrsRequestResource()) { … }` since we never end up talking to DRS, but calling this out.

Reply:
I haven't looked at this code in a while, but maybe the correct thing is to put `try (DrsRequestResource r = new DrsRequestResource())` inside `lookupSnapshotForDRSObject()`

Reply:
I'm hesitant to move it to lookupSnapshotForDRSObject() because then it would seem like we are only monitoring the number of requests because of the snapshot lookup. My understanding is that we want this on the whole drs request to prevent getting overwhelmed with calls to our drs endpoints. 

To @okotsopoulos 's question, I think it's reasonable to also wrap the options endpoint in a drs request. It's another endpoint that a service could really hammer on, but given that it's a much lighter weight method, it might not be necessary. Thoughts, @nmalfroy ?

Reply:
I think it makes sense to include the options endpoint in the cache since I agree it could get hammered and I think it does still require a DB lookup to get the snapshot's allowed auth (I think this is the question you were asking?)

Reply:
It's not the cache - we wrap some of the other DrsService methods in the DrsRequestResource that limits how many drs requests can be made at once - https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/filedata/DrsService.java#L138

Reply:
Oh right, sorry.  I haven't had a chance to look too closely.  Yeah...that might be worth including it too

Reply:
Thanks folks, I wrapped the body of this method with a DrsRequestResource.

Reply:
> I'm hesitant to move it to lookupSnapshotForDRSObject() because then it would seem like we are only monitoring the number of requests because of the snapshot lookup. My understanding is that we want this on the whole drs request to prevent getting overwhelmed with calls to our drs endpoints.

Looking at our code, `lookupSnapshotForDRSObject()` is currently called in two places where `DrsRequestResource` is used. This would now be a third case. I guess the downside is that the other two uses go on to call `lookupDRSObjectAfterAuth()` and this doesn't do that, so the "lock" that `DrsRequestResource` provides would be released earlier (before the lookup call) if it was moved. I doubt that would make a big difference since `DrsRequestResource` is still gating the number of requests occurring, but it would release a little earlier.